control/state.py

```diff
@@ -229,53 +277,67 @@ def get_state(self) -> Dict[str, str]:
             omap_dict = dict(i)
         return omap_dict
 
-    def _add_key(self, key: str, val: str):
+    def _add_key(self, key: str, val: str, status: RequestStatus):
         """Adds key and value to the OMAP."""
-        try:
-            version_update = self.version + 1
-            with rados.WriteOpCtx() as write_op:
-                # Compare operation failure will cause write failure
-                write_op.omap_cmp(self.OMAP_VERSION_KEY, str(self.version),
-                                  rados.LIBRADOS_CMPXATTR_OP_EQ)
-                self.ioctx.set_omap(write_op, (key,), (val,))
-                self.ioctx.set_omap(write_op, (self.OMAP_VERSION_KEY,),
-                                    (str(version_update),))
-                self.ioctx.operate_write_op(write_op, self.omap_name)
-            self.version = version_update
-            self.logger.debug(f"omap_key generated: {key}")
-        except Exception as ex:
-            self.logger.error(f"Unable to add key to omap: {ex}. Exiting!")
+        version_update = self.version + 1
+        for i in range(self.state_update_retries):
+            try:
+                with rados.WriteOpCtx() as write_op:
+                    # Compare operation failure will cause write failure
+                    write_op.omap_cmp(self.OMAP_VERSION_KEY, str(self.version),
+                                      rados.LIBRADOS_CMPXATTR_OP_EQ)
+                    self.ioctx.set_omap(write_op, (key,), (val,))
+                    self.ioctx.set_omap(write_op, (self.OMAP_VERSION_KEY,),
+                                        (str(version_update),))
+                    self.ioctx.operate_write_op(write_op, self.omap_name)
+            except Exception as ex:  # Update and retry on failure
+                self.update_trigger.set()
+                time.sleep(1)
```

Comment:
Sleeping 1 sec may lead to unnecessary delays. This could be improved by introducing yet another event variable. It could wait on it here until the update thread sets it after completing update.